File: 프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.java
import java.util.*;

class Solution {
    public String solution(String s) {
        StringBuilder result = new StringBuilder();
        String[] words = s.split(" ") + " ";
        System.out.println(words[2]);
        
        for (String word : words) {
            if (!word.equals(""))
                for (int idx = 0; idx < word.length(); idx++) {
                    if (word.charAt(idx) % 2 == 0) {
                        result.append(word.charAt(idx));
                    }
                    else result.append(word.charAt(idx));
                }
            else continue;
            
        }
        return result.toString();
    }
}

File: 프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.js
function solution(s) {
    let result = "";
    let count1 = 0;
    const word = s.trim().replace(/\s+/g, ' ').split(" ");
    
    for (let i = 0; i < s.length; i++) {
        if (s[i] === " ") {
            result += " ";
        }
        else {
            let newword = "";
            for (let j = 0; j < word[count1].length; j++) {
                if (j % 2 === 0) newword += word[count1][j].toUpperCase();
                else if (j % 2 === 1) newword += word[count1][j].toLowerCase();
            }
            result += newword;
            i += newword.length - 1;
            count1++;
        }
    }
        
    return result;
}